### DISCUSSION TITLE
packaging Apk with latest Adobe air sdk + latest Distriqt Anes

### DISCUSSION THREAD

**Discussion by ronyarbiv** [airsdk/Adobe-Runtime-Support#1924] [General]
Hi,

I am using the latest Air sdk, Latest Distriqt Anes.
I am using Intellij latest version.

From Intellij ->Build -> Packaging Air application I am manage to pack the apk.
The problem is that when i extract the apk to see the content, it seems like it contain only the swc of every Ane.
the apk is 10 mb less the expected.

several questions-

1.Can i still package apps with intellij? (for simplisity i have temporarly signed the app with temporarly key and not my p12 key, and remove 1.8 java from the machine and from the enviromental variabel, i have installed android studio and android sdk)
2.The app is really old (but it still get alot of downloads every day) so it uses 1024 p12 key. so as far that i know inorder tho pack it i need to use the old java 1.8. i don't know if it means that i will not be able to combine android 31 support with the old java and the atest anes?

**Comment by ajwfrost** [comment]
If the ANEs use Java, then the Java files are just being linked into the application's dex code and we're then not bothering to pack up the .jar files any more (not actually needed). Likewise any resources that the ANEs contain will be compiled into the resource bundle and omitted from the apk. So all that's actually needed is the library swc file so that the AIR runtime can pick up the ANE's interface definitions.

For building if you're using a 1024-bit key, you may need to do it in two stages. You will need to use Java 11 or similar, and the new Gradle-based build mechanism, to create your packages; you can then re-sign the resulting output using Java 1.8 and the older Android tools (apksigner). Or if you're generating an AAB bundle you don't actually need to sign it when building with AIR and can just jar-sign it later on.

But there's then the difference between the signing key and the upload key .. for details see https://airsdk.dev/docs/tutorials/platform/android/packaging-android-app-bundles#signing-and-certificates

thanks

**Reply by ronyarbiv** [reply]
Thank you very much.
The app is crashing on startup so i thought the reason was that it didn't pack all it needed. 
So i will direct my searching to other directions. 

thanks

**Reply by ajwfrost** [reply]
If you can check the logcat output it may have a clue as to what is causing the crash-at-startup. Sometimes if we don't get the right architectures then this might happen (although that shouldn't be the case if you're using the Distriqt ANEs, they support all the possible architectures)

**Reply by marchbold** [reply]
@ronyarbiv One of the many improvements Harman have made! :) Removing all the redundant files that use to get packaged from an ANE. 

If your app is crashing on startup definitely check the device log if often points to the issue. Most commonly a crash at startup is due to something amiss in your manifest additions, eg referencing a class in the manifest that doesn't exist in your application package.

**Reply by marchbold** [reply]
Regarding your key, I suggest you create an upload certificate separate from your signing certificate and upload your old signing certificate to google play. That way you can use a new 2048 bit cert with java 11 to sign your app and dev, while handing your old certificate to google to use to release your app.

More on the process here:

https://airsdk.dev/docs/tutorials/platform/android/packaging-android-app-bundles

**Reply by ronyarbiv** [reply]
thank you so much!
Thats indeed amazing. it saves 10 mb that will be filled with more games for my users. 
I am really happy that i have asked, as i will change my investigation to another direction.
the only ane that is not from Distriqt is ezspeech, i will try to remove it and check if thats the problem. 

also started with a new empty project and using distriqt apm which sounds amazing. when i will succeed with the apm on the empty project, i want to try and add the apm to my 9 year old app. 

I will try more with the key, i have tried several instructions but nothing worked yet.  i will thry the link you have sent here. 

I must say that on the last 9 years with adobe, i feel now safer and happier then any onther time - Harman, Distriqt, Starling are all working so efficiently and innovative and help so fast. that give me as a developer all that i need so i can focus on just developing my games. 

Best wishes, 
Rony 

I
